src/Makefile.test.include

```diff
@@ -416,6 +416,31 @@ if EMBEDDED_UNIVALUE
 	$(AM_V_at)$(MAKE) $(AM_MAKEFLAGS) -C univalue check
 endif
 
+if ENABLE_BENCH
+if ENABLE_QT_TESTS
+check-valgrind: $(TEST_BINARY) $(BENCH_BINARY) test_bitcoin_qt
+else
+check-valgrind: $(TEST_BINARY) $(BENCH_BINARY)
+endif
+else
+if ENABLE_QT_TESTS
+check-valgrind: $(TEST_BINARY) test_bitcoin_qt
+else
+check-valgrind: $(TEST_BINARY)
+endif
+endif
+	@echo "Using the valgrind memory error detector: expect a 10-50x slowdown and ~2x memory usage. valgrind 3.14 or later required."
+	@echo "Running $(TEST_BINARY) under valgrind -- this will take a a while..."
+	valgrind --suppressions=$(top_builddir)/contrib/valgrind.supp --gen-suppressions=all --exit-on-first-error=yes --error-exitcode=1 --quiet $(TEST_BINARY)
```

Comment:
To have valgrind make it through the unit tests, I had to debug rather extensively my suppressions file by running  `valgrind --gen-suppressions=all --verbose --exit-on-first-error=yes --error-exitcode=1 --suppressions=contrib/valgrind.supp src/test/test_bitcoin --log_level=test_suite` and adding suppressions. It takes 60-90 minutes for a run, so it's a long process..

- Is the suppressions file for a project supposed to be portable across platforms, or is it normal for it to be a config adventure for each user?

- I found running valgrind directly more informative for debugging and feedback, particularly with flags like `--verbose` and `--log_level=test_suite`. I did not try running `make check-valgrind` with those options yet but wonder if it is a convenience feature, or adding other value that I'm misunderstanding? I do intend to retest and re-review.

Reply:
@jonatack 

I don't think you're doing anything wrong: Valgrind is super slow :)

> Is the suppressions file for a project supposed to be portable across platforms, or is it normal for it to be a config adventure for each user?

I think the current set of suppressions have been tested and updated by at least me and @MarcoFalke for various configurations, but more additions are welcome :)  Please submit your local changes upstreams :)

The suppressions are not necessarily portable across distributions, or across versions of the same distribution.

> I found running valgrind directly more informative for debugging and feedback, particularly with flags like --verbose and --log_level=test_suite. I did not try running make check-valgrind with those options yet but wonder if it is a convenience feature, or adding other value that I'm misunderstanding? I do intend to retest and re-review.

I usually run with `--log_level=test_suite` when debugging a found error, but I'm afraid enabling it by default for `make check-valgrind` would make the output too verbose. Generally I think `check`-type commands should be silent unless an error is found, and hopefully the most common scenarios is that of no Valgrind errors :)

Reply:
Retested this PR and wonder if it would not be better to remove the --quiet flag... `make check` isn't quiet either, `make check-valgrind` takes a long time (~90 minutes) to run so some output is reassuring, and the additional output is useful:
```
Using the valgrind memory error detector: expect a 10-50x slowdown and ~2x memory usage. valgrind 3.14 or later required.
Running test/test_bitcoin under valgrind -- this will take a a while...
valgrind --suppressions=../contrib/valgrind.supp --gen-suppressions=all --exit-on-first-error=yes --error-exitcode=1 test/test_bitcoin
==36831== Memcheck, a memory error detector
==36831== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==36831== Using Valgrind-3.16.0.GIT and LibVEX; rerun with -h for copyright info
==36831== Command: test/test_bitcoin
==36831== 
Running 391 test cases...
```